docs/get-started/deployment-and-usage.md

```diff
@@ -426,29 +424,178 @@ configuration for enabling TLS authentication.
+created and stored as secrets within the NFD namespace.
+
+Create a CA certificate
+
+```bash
+
+openssl genrsa -out ca.key 4096
+openssl req -x509 -new -nodes -key ca.key -subj "/CN=nfd-ca" \
+        -days 10000 -out ca.crt
+
+```
+
+Create a certificate for the nfd-master
+
+```bash
+
+openssl genrsa -out nfd-master.key 4096
+
+cat <<EOF > nfd-master.conf
+[ req ]
+default_bits = 4096
+prompt = no
+default_md = sha256
+req_extensions = req_ext
+distinguished_name = dn
+
+[ dn ]
+C = XX
+ST = some-state
+L = some-city
+O = some-company
+OU = node-feature-discovery
+CN = nfd-master
+
+[ req_ext ]
+subjectAltName = @alt_names
+
+[ alt_names ]
+DNS.1 = nfd-master
+DNS.2 = nfd-master.node-feature-discovery.svc.cluster.local
+DNS.3 = localhost
+
+[ v3_ext ]
+authorityKeyIdentifier=keyid,issuer:always
+basicConstraints=CA:FALSE
+keyUsage=keyEncipherment,dataEncipherment
+extendedKeyUsage=serverAuth,clientAuth
+subjectAltName=@alt_names
+EOF
+
+openssl req -new -key nfd-master.key -out nfd-master.csr -config nfd-master.conf
+```
+
+Create certificate for nfd-worker (and, if needed, nfd-topology-updater)
+
+```bash
+
+openssl genrsa -out nfd-worker.key 4096
+
+cat <<EOF > nfd-worker.conf
+[ req ]
+default_bits = 4096
+prompt = no
+default_md = sha256
+req_extensions = req_ext
+distinguished_name = dn
+
+[ dn ]
+C = XX
+ST = some-state
+L = some-city
+O = some-company
+OU = some-org-unit
+CN = nfd-worker
+
+[ req_ext ]
+subjectAltName = @alt_names
+
+[ alt_names ]
+DNS.1 = nfd-worker
+DNS.2 = nfd-worker.node-feature-discovery.svc.cluster.local
+
+[ v3_ext ]
+authorityKeyIdentifier=keyid,issuer:always
+basicConstraints=CA:FALSE
+keyUsage=keyEncipherment,dataEncipherment
+extendedKeyUsage=serverAuth,clientAuth
+subjectAltName=@alt_names
+EOF
+
+openssl req -new -key nfd-worker.key -out nfd-worker.csr -config nfd-worker.conf
+
+# ...
+
+openssl genrsa -out nfd-topology-updater.key 4096
+sed -e 's/worker/topology-updater/g' < nfd-worker.conf > nfd-topology-updater.conf
+openssl req -new -key nfd-topology-updater.key -out nfd-topology-updater.csr -config nfd-topology-updater.conf
+
+
+```
+
+Now, sign the certificates with the CA created earlier.
+
+```bash
+for cert in nfd-master nfd-worker nfd-topology-updater; do
+  echo signing $cert
+  openssl x509 -req -in $cert.csr -CA ca.crt -CAkey ca.key \
+    -CAcreateserial -out $cert.crt -days 10000 \
+    -extensions v3_ext -extfile $cert.conf
+done
+```
+
+Finally, turn these certificates into secrets.
+
+```bash
+for cert in nfd-master nfd-worker nfd-topology-updater; do
+  echo creating secret for $cert in node-feature-discovery namespace
+  cat <<EOF | kubectl create -n node-feature-discovery -f -
+---
+apiVersion: v1
+kind: Secret
+type: kubernetes.io/tls
+metadata:
+  name: ${cert}-cert2
+data:
+  ca.crt: $( cat ca.crt | base64 -w 0 )
+  tls.crt: $( cat $cert.crt | base64 -w 0 )
+  tls.key: $( cat $cert.key | base64 -w 0 )
+EOF
```

Comment:
Would it be possible to use `kubectl create secret tls ...` instead here (I don't remember the details of that on top of my head)? Would greatly simplify generation of the secret

Reply:
No - the "kubectl create secret tls" can only handle the tls.crt and tls.key, not the ca.crt.  I tried it and you end up with a race condition where the secret is created but isn't complete.  The pods start and crash immediately because ca.crt isn't there.

The underlying problem is that cert-manager creates the secrets with ca.crt embedded in the TLS secret, and we either need our manual process to duplicate that, or have different handling for the two situations (e.g. have the ca.crt in a configMap)

Reply:
K, thanks for explanation. Let's go with the yaml, then. A patch to kubectl to make it possible to add ca cert, too, would be nice addtion :)

Reply:
Agree -- it looks a little awkward to create the secret this way but it's 100% better than creating an incomplete one and then adding the extra element later.